torch/distributed/_checkpoint/storage_reader.py

```diff
@@ -0,0 +1,98 @@
+import abc
+import operator
+import os
+import pickle
+from typing import List, Optional, cast
+
+import torch
+from torch import Tensor
+from torch.futures import Future
+
+from .metadata import BytesReadRequest, Metadata, TensorReadRequest
+
+
+class StorageReader(abc.ABC):
+    """
+    Interface to read from the underlying storage system.
+    """
+
+    @abc.abstractmethod
+    def read_bytes(self, requests: List[BytesReadRequest]) -> Future[None]:
```

Comment:
I think unless we know what that something clearly is (regardless of the type) we should not change the signature.

Since those values would be implementation dependent on the subclass of StorageReader, users could just `cast(Future[Something], reader.read_bytes(...))`.

Reply:
IIRC,  Future[Any]  does not pass the linter